### PR TITLE
Added a docstring for torch.Size.numel.

### LINKED ISSUE #61231: Document torch.Size and its methods (like torch.Size.numel)
## 📚 Documentation

I see `torch.Size` and `torch.Size.numel` are undocumented. This issue is a request to document them, since I have a situation where I only have `tensor.Size` and I'm not sure if I can reliably use `shape.numel()`, hence the uncertainly.

Thank you!

cc @brianjo @mruberry

### LINKED ISSUE #124167: numel() works differently for torch.Tensor and torch.Size inputs, not documented
### 📚 The doc issue

`numel()` works differently for `torch.Tensor` and `torch.Size` inputs. This is not mentioned in the [documentation](https://pytorch.org/docs/stable/generated/torch.numel.html).

**Example:**
```
import torch
a = torch.Tensor(3, 4)
print(a.numel(), a.shape.numel())
```
Output:
12
12

As I was reading code I expected shape.numel() to return the number of dimensions, so I was surprised that it works this way. I understand that this is a feature but I think this should be mentioned in the documentation.

### Suggest a potential alternative/fix

It could be mentioned in the documentation if `numel()` is called on a `Size `object it returns the number of items in a that sized `Tensor`.

cc @svekars @brycebortree

### PR DESCRIPTION
**Pull request by tringwald**
Fixes #61231. Fixes #124167.

This PR documents a rather long-standing issue w.r.t. unexpected behavior of `torch.Size.numel`, first reported almost 5 years ago.

### CODE REVIEW

File: torch/_size_docs.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+add_docstr_all(
+    "numel",
+    """
+numel() -> int
+
+Returns the number of elements a :class:`torch.Tensor` with the given size would contain.
+
+More formally, for a tensor ``x = tensor.ones(10, 10)`` with size ``s = torch.Size([10, 10])``,
+``x.numel() == x.size().numel() == s.numel() == 100`` holds true.
```
Comment (janeyx99) [inline torch/_size_docs.py:19]: oh wow this is actually true, contrary to https://github.com/pytorch/pytorch/pull/61714#issuecomment-885404561 now. I wonder when this changed

**Review by janeyx99 — COMMENTED** [review]
image
I cannot find the added docs when clicking on the python docs preview here and searching. Not sure if the docs got added properly?

Also, in the future, it would be nice in docs updates to include a screenshot of what the before and after of the docs looked like!

File: torch/_size_docs.py (hunk 1 above)
Comment (soulitzer) [reply torch/_size_docs.py:19]: I think that comment was just talking about how `torch.Size.numel()` and `torch.numel()` are distinct APIs that each need to have their own docstrings.

Comment (tringwald) [reply torch/_size_docs.py:19]: @soulitzer is right. The actual behavior was always like that, the comment just says those two functions are unrelated.

**Comment by tringwald** [comment]
I added a new docs page for `torch.Size` and added it to the sidebar as most other top-level classes/modules.

**Review by janeyx99 — APPROVED** [review]
IMG_4846

Looks great! One formatting q: do you know what the slashes are in index and count?

**Comment by tringwald** [comment]
> Looks great! One formatting q: do you know what the slashes are in index and count?

Those functions are inherited from `tuple`. The `/` divides the function parameters into positional-only and positional/keyword arguments.

**Pull request closed without merging**